### PR TITLE
Minor performance improvement for supermatter cascade

### PR DESCRIPTION
**Pull request by gbasood**
Because why go through all the bother of animating a thing and changing a tile to an identical tile if we don't need to

### CODE REVIEW

**Comment by clusterfack** [comment]
Pretty sure this wouldn't happen anyway unless there were two independent bluespace events colliding

**Comment by gbasood** [comment]
If two bluespace tiles are created adjacent together around the same time, it sure can.
Or, you know, if a new tile tries to expand onto the tile that created it.

**Comment by gbasood** [comment]
Just made another small improvement for that second instance.
Except I fucked it up hold on

File: code/game/gamemodes/endgame/supermatter_cascade/blob.dm [hunk 1]
```diff
@@ -58,6 +61,9 @@
 					A = null
 				tcheck(80,1)
 			T.ChangeTurf(type)
+			var/turf/unsimulated/wall/supermatter/SM = T
+			if(SM.avail_dirs)
+				SM.avail_dirs -= get_dir(T, src)
```
Comment (gbasood) [inline code/game/gamemodes/endgame/supermatter_cascade/blob.dm]: Now new supermatter tiles won't check spawn a new tile on the tile that spawned them

File: code/game/gamemodes/endgame/supermatter_cascade/blob.dm [hunk 2]
```diff
@@ -58,6 +61,9 @@
...
 
 	if((spawned & (NORTH|SOUTH|EAST|WEST)) == (NORTH|SOUTH|EAST|WEST))
```
Comment (clusterfack) [inline code/game/gamemodes/endgame/supermatter_cascade/blob.dm]: This line does nothing you know

Comment (gbasood) [inline code/game/gamemodes/endgame/supermatter_cascade/blob.dm]: The if((spawned thing? Yeah I was wondering about that

**Pull request merged**